### PR TITLE
Use first image (after refactor)

### PR DESCRIPTION
**Pull request by TheTripleV**
This PR is functionally the same as the description in #18 but is applied after the refactor.
Could you take a look at it, @tcmetzger ?

### CODE REVIEW

**Comment by tcmetzger** [comment]
Absolutely - just give me few days!

**Comment by tcmetzger** [comment]
One small thing I noticed right away: in readme.md, line 16, there should be quotation marks around `sphinxext.opengraph` (should be a list of strings).

File: sphinxext/opengraph/__init__.py [hunk 1]
```diff
@@ -55,19 +70,27 @@ def get_tags(
...
     # image tag
-    # Get the image from the config
+    # Get basic values from config
     image_url = config["ogp_image"]
+    ogp_use_first_image = config["ogp_use_first_image"]
+    ogp_image_alt = config["ogp_image_alt"]
+
+    if ogp_use_first_image:
+        first_image = doctree.next_node(nodes.image)
+        if Path(first_image.get("uri", "")).suffix[1:].lower() in IMAGE_MIME_TYPES:
```
Comment (tcmetzger) [inline sphinxext/opengraph/__init__.py:80]: I believe this line throws an error (`'NoneType' object has no attribute 'get'`) under the following conditions:
* `ogp_use_first_image ` is set to True in conf.py.
* there is a .rst file that does not include any images.

According to README.md, the expected behaviour in a case like this is to use `ogp_image`.

**Comment by tcmetzger** [comment]
Those two issues are the only two things I was able to see. 
Thanks for your work on this very helpful extension!

**Review by tcmetzger — CHANGES_REQUESTED** [review]
Changes I recommend:
* add quotation marks to README.md, line 16
* handle a situation where `ogp_use_first_image` is enabled but no image is present

**Comment by TheTripleV** [comment]
I addressed the 2 issues and added a test case to cover

> situation where ogp_use_first_image is enabled but no image is present

**Review by tcmetzger — APPROVED** [review]
LGTM!

**Review by Daltz333 — APPROVED** [review]

**Pull request merged**